packages/blocks/src/page-block/edgeless/components/toolbar/edgeless-toolbar.ts

```diff
@@ -159,9 +200,156 @@ export class EdgelessToolbar extends WithDisposable(LitElement) {
     }
   };
 
+  protected override updated(changedProperties: PropertyValues) {
+    const { type } = this.edgelessTool;
+    if (changedProperties.has('_currentFrameIndex') && type === 'present') {
+      const frames = this.edgeless.surface.getElementsByType('frame');
+      const current = this._currentFrameIndex;
+      const viewport = this.edgeless.surface.viewport;
+      const frame = frames[current];
+      if (frame) {
+        const bound = Bound.deserialize(frame.xywh);
+        viewport.setViewportByBound(bound, [40, 60], true);
+      }
+    }
+  }
+
+  private get prensentContent() {
```

Comment:
The `_foo` convention also applies for methods.

Reply:
Also, I think we should extract the frame navigator toolbar as a component.

Reply:
At first, I also was intended to make a new component for frame navigator.
But in this way, I can't reuse it's parent (toolbar container) style because lit doesn't have sort of `Fragment`.

Reply:
> The `_foo` convention also applies for methods.
> 
> But here, this variable and the `defaultContent` should both be extracted as pure template function (with `MyTemplate` naming convention), instead of private methods.

defaultContent depends on many states of `EdgelessToolbar`, is it better to be extracted as a pure template function?